clang/test/SemaHLSL/BuiltIns/asdouble-errors.hlsl

```diff
@@ -0,0 +1,16 @@
+// RUN: %clang_cc1 -finclude-default-header -triple dxil-pc-shadermodel6.6-library %s -emit-llvm-only -disable-llvm-passes -verify
+
+double test_too_few_arg() {
+  return __builtin_hlsl_asdouble();
+  // expected-error@-1 {{too few arguments to function call, expected 2, have 0}}
+}
+
+double test_too_few_arg_1(uint p0) {
+  return __builtin_hlsl_asdouble(p0);
+  // expected-error@-1 {{too few arguments to function call, expected 2, have 1}}
+}
+
+double test_too_many_arg(uint p0) {
+  return __builtin_hlsl_asdouble(p0, p0, p0);
+  // expected-error@-1 {{too many arguments to function call, expected 2, have 3}}
+}
```

Comment:
I feel like adding some tests to check for type mismatch would be useful

Reply:
Is there a type that you have in mind? Most types will be implicitly converted here, I added the custom struct type to illustrate that.

Reply:
Any type really, what we really care about here is making sure the error is meaningful to the user. I usually write such tests with `half`, `float` or `bool` if the former are implicitly cast.